nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-audit/src/test/java/org/nuxeo/elasticsearch/TestESHistoryProvider.java

```diff
@@ -191,152 +291,37 @@ protected void createTestEntries() throws Exception {
-        log.trace("Proxy doc history");
-        entries.forEach(entry -> log.trace("LogEntry: {}", entry));
-
-        // 19 - 5 updates + create + proxyPublished
-        int proxyEntriesCount = 19 - 5 + 1 + 1;
-        assertEquals(proxyEntriesCount, entries.size());
-
-        assertEquals(Long.valueOf(startIdx).longValue(), entries.get(0).getId());
-        assertEquals(Long.valueOf(startIdx + proxyEntriesCount + 1).longValue(),
-                entries.get(proxyEntriesCount - 1).getId());
-
-        // Get version 1 history
-        pp = pps.getPageProvider("DOCUMENT_HISTORY_PROVIDER", si, Long.valueOf(20), Long.valueOf(0), new HashMap<>(),
-                versions.get(0));
-        pp.setSearchDocumentModel(searchDoc);
-        entries = (List<LogEntry>) pp.getCurrentPage();
-
-        log.trace("Verion {} doc history", () -> versions.get(0).getVersionLabel());
-        entries.forEach(entry -> log.trace("LogEntry: {}", entry));
-
-        // creation + 5 updates + update + checkin + created
-        int version1EntriesCount = 1 + 5 + 1 + 1 + 1;
-        if (version1EntriesCount == entries.size()) {
-            assertEquals(Long.valueOf(startIdx).longValue(), entries.get(0).getId());
-            assertEquals(Long.valueOf(startIdx + version1EntriesCount - 1).longValue(),
-                    entries.get(version1EntriesCount - 1).getId());
-        } else {
-            // because update even may be 1ms behind checkin/created !
-            assertEquals(version1EntriesCount - 1, entries.size());
-        }
-
-        // get version 2 history
-        pp = pps.getPageProvider("DOCUMENT_HISTORY_PROVIDER", si, Long.valueOf(20), Long.valueOf(0), new HashMap<>(),
-                versions.get(1));
-        pp.setSearchDocumentModel(searchDoc);
-
-        entries = (List<LogEntry>) pp.getCurrentPage();
-
-        log.trace("Verion {} doc history", () -> versions.get(1).getVersionLabel());
-        entries.forEach(entry -> log.trace("LogEntry: {}", entry));
+    protected PageProvider<LogEntry> getPageProvider(String name, int pageSize, int currentPage, Object... parameters) {
+        final List<SortInfo> sorters = List.of(new SortInfo("id", true));
+        @SuppressWarnings("unchecked")
+        PageProvider<LogEntry> pageProvider = (PageProvider<LogEntry>) pageProviderService.getPageProvider(name,
+                sorters, Long.valueOf(pageSize), Long.valueOf(currentPage), Map.of(), parameters);
+        return pageProvider;
+    }
 
-        // creation + 5x2 updates + checkin/update + checkin + created
-        int versin2EntriesCount = 1 + 5 * 2 + 1 + 1 + 1 + 1;
-        assertEquals(versin2EntriesCount, entries.size());
-        assertEquals(Long.valueOf(startIdx).longValue(), entries.get(0).getId());
-        assertEquals(Long.valueOf(startIdx + versin2EntriesCount).longValue(),
-                entries.get(versin2EntriesCount - 1).getId());
+    protected void waitUntil(long millis) throws InterruptedException {
```

Comment:
What is the point to hide the Thread.sleep here? the thread is sleeping for an amount of time not waiting until a date.

Reply:
yep initially when i stared the clean up my idea is to replace the `Thread.sleep` by `awaitility` which is used now on master. but unfortunately the tests fails when i made some tests. and i just extract the method, and change using `TimeUnit.MILLISECONDS.sleep(millis)` which remove the waring that i had https://rules.sonarsource.com/java/RSPEC-2925

For the name i agree it don't reflect exactly  what it did. i will rollback this part, i mean i will use directly `TimeUnit.MILLISECONDS.sleep(millis)`